plugins/autopeering/local/parameters.go

```diff
@@ -1,23 +1,34 @@
 package local
 
 import (
-	flag "github.com/spf13/pflag"
+	"github.com/iotaledger/hive.go/configuration"
 )
 
-const (
-	// CfgBind defines the config flag of the network bind address.
-	CfgBind = "network.bindAddress"
-	// CfgExternal defines the config flag of the network external address.
-	CfgExternal = "network.externalAddress"
-	// CfgPort defines the config flag of the autopeering port.
-	CfgPort = "autopeering.port"
-	// CfgSeed defines the config flag of the autopeering private key seed.
-	CfgSeed = "autopeering.seed"
-)
+// ParametersDefinitionLocal contains the definition of configuration parameters used by the local peer.
+type ParametersDefinitionLocal struct {
+	// Port defines the config flag of the autopeering port.
+	Port int `default:"14626" usage:"UDP port for incoming peering requests"`
+
+	// Seed defines the config flag of the autopeering private key seed.
+	Seed string `usage:"private key seed used to derive the node identity; optional base58 or base64 encoded 256-bit string. Prefix with 'base58:' or 'base64', respectively"`
+}
+
+// ParametersDefinitionNetwork contains the definition of configuration parameters used by the local peer's network.
+type ParametersDefinitionNetwork struct {
+	// BindAddress defines the config flag of the network bind address.
+	BindAddress string `default:"0.0.0.0" usage:"bind address for global services such as autopeering and gossip"`
+
+	// ExternalAddress defines the config flag of the network external address.
+	ExternalAddress string `default:"auto" usage:"external IP address under which the node is reachable; or 'auto' to determine it automatically"`
+}
+
+// ParametersLocal contains the configuration parameters of the local peer.
+var ParametersLocal = ParametersDefinitionLocal{}
+
+// ParametersNetwork contains the configuration parameters of the local peer's network.
+var ParametersNetwork = ParametersDefinitionNetwork{}
 
 func init() {
-	flag.String(CfgBind, "0.0.0.0", "bind address for global services such as autopeering and gossip")
-	flag.String(CfgExternal, "auto", "external IP address under which the node is reachable; or 'auto' to determine it automatically")
-	flag.Int(CfgPort, 14626, "UDP port for incoming peering requests")
-	flag.String(CfgSeed, "", "private key seed used to derive the node identity; optional base58 or base64 encoded 256-bit string. Prefix with 'base58:' or 'base64', respectively")
+	configuration.BindParameters(&ParametersLocal, "autopeering")
```

Comment:
I noticed several parameter instances are mapped to "autopeering" (e.g. see also discovery/parameters.go). What happens if we incidently create two parameters with the same name. Do they write to the same parameter in "autopeering" ?

Reply:
Yeah this is not optimal. However, we didn't want to break the interface with this PR. I think cleaning up the configuration parameters should be done with https://github.com/iotaledger/goshimmer/issues/1507

Reply:
Ok